src/helm/benchmark/runner.py

```diff
@@ -138,10 +143,24 @@ def run_one(self, run_spec: RunSpec):
         adapter: Adapter = AdapterFactory.get_adapter(run_spec.adapter_spec, self.tokenizer_service)
 
         instances: List[Instance]
-        if not self.skip_instances:
-            # Create the instances of the scenario
-            with htrack_block("scenario.get_instances"):
-                instances = scenario.get_instances()
+        if self.skip_instances:
+            instances = []
+        else:
+            if self.read_instances_from_file:
+                with open(os.path.join(scenario.output_path, "instances.json")) as f:
+                    json_instances: List[Dict[str, Any]] = json.load(f)
+                instances = [dacite.from_dict(Instance, instance) for instance in json_instances]
+            else:
+                # Create the instances of the scenario
+                with htrack_block("scenario.get_instances"):
+                    instances = scenario.get_instances()
+                # Save instances to file
+                write(
+                    os.path.join(scenario.output_path, "instances.json"),
+                    json.dumps([asdict_without_nones(instance) for instance in instances], indent=2),
+                )
```

Comment:
I wonder if this write should also be controlled by a flag... since for normal users, this file gets written out and then never gets used.

Reply:
Done.